### PR TITLE
Make default param values be picked up as such

### LINKED ISSUE #1788: Implicitly typed Optional parameter causes Resolver Error
```
Function Foo(Optional var)
End Function
```
```
Function Foo(Optional var = "bar")
End Function
```

### PR DESCRIPTION
**Pull request by Hosch250**
Close #1788

### CODE REVIEW

File: Rubberduck.Parsing/Grammar/VBAParser.g4 [hunk 1]
```diff
@@ -530,9 +530,18 @@ baseType : BOOLEAN | BYTE | CURRENCY | DATE | DOUBLE | INTEGER | LONG | LONGLONG
 
 comparisonOperator : LT | LEQ | GT | GEQ | EQ | NEQ | IS | LIKE;
 
-complexType : expression;
+complexType :
```
Comment (autoboosh) [inline Rubberduck.Parsing/Grammar/VBAParser.g4]: PC crashed. Looks fine then I guess? Haven't looked at the resolver part though.

File: Rubberduck.Parsing/Symbols/IdentifierReferenceResolver.cs [hunk 2]
```diff
@@ -555,13 +555,13 @@ public void Resolve(VBAParser.AsTypeClauseContext context)
             {
                 // Fixed-Length strings can have a constant-name as length that is a simple-name-expression that also has to be resolved.
                 var length = context.fieldLength();
-                if (context.fieldLength() != null)
+                if (context.fieldLength() != null && context.fieldLength().identifierValue() != null)
                 {
-                    ResolveDefault(context.fieldLength().expression());
+                    ResolveDefault(context.fieldLength().identifierValue());
```
Comment (retailcoder) [inline Rubberduck.Parsing/Symbols/IdentifierReferenceResolver.cs]: @autoboosh this looks good to me, the `fieldLength` can't be `const1 * const2` if `identifierValue() != null`, right?

File: Rubberduck.Parsing/Symbols/IdentifierReferenceResolver.cs [hunk 3]
```diff
@@ -555,13 +555,13 @@ public void Resolve(VBAParser.AsTypeClauseContext context)
...
                 }
                 return;
             }
-            ResolveType(asType.complexType().expression());
+            ResolveType(asType.complexType());
```
Comment (retailcoder) [inline Rubberduck.Parsing/Symbols/IdentifierReferenceResolver.cs]: What's the difference between `ResolveType(asType.complexType());` and `ResolveType(asType.complexType().expression());`?

File: Rubberduck.Parsing/Symbols/IdentifierReferenceResolver.cs (hunk 2 above)
Comment (Hosch250) [inline Rubberduck.Parsing/Symbols/IdentifierReferenceResolver.cs]: If `identifierValue()` is null, it is a numeric literal.

File: Rubberduck.Parsing/Symbols/IdentifierReferenceResolver.cs (hunk 3 above)
Comment (Hosch250) [inline Rubberduck.Parsing/Symbols/IdentifierReferenceResolver.cs]: `complexType()` does not contain `expression()` anymore.

**Pull request merged**